### PR TITLE
Feat/network js fix

### PR DESCRIPTION
**Pull request by sneakzttv**
Automatically writes the network.js files on ProjectID save so the users no longer need to!

### CODE REVIEW

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs [hunk 1]
```diff
@@ -125,4 +137,30 @@ void OnGUI()
...
+        // declares paths to write our javascript files to
+        string path1 = "Assets/WebGLTemplates/Web3GL-2020x/network.js";
+        string path2 = "Assets/WebGLTemplates/Web3GL-MetaMask/network.js";
+
+        // writes data to the webgl default network file
+        StreamWriter writer1 = new StreamWriter(path1, false);
+        writer1.WriteLine("//You can see a list of compatible EVM chains at https://chainlist.org/");
+        writer1.WriteLine("window.networks = [");
+        writer1.WriteLine("  {");
+        writer1.WriteLine("id: " + PlayerPrefs.GetString("ChainID") + ",");
```
Comment (Arshia001) [inline src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:152]: These need to be indented with 4 spaces.

Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:152]: tab only gave me 2 for some reason, odd

Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:152]: moment

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs [hunk 2]
```diff
@@ -125,4 +137,30 @@ void OnGUI()
         }
         GUILayout.Label("Reminder: Your ProjectID Must Be Valid To Save & Build With Our SDK. You Can Register For One On Our Website At Dashboard.Gaming.Chainsafe.io", EditorStyles.label);
     }
+
+    public void WriteNetworkFile()
+    {
+        // declares paths to write our javascript files to
+        string path1 = "Assets/WebGLTemplates/Web3GL-2020x/network.js";
```
Comment (Arshia001) [inline src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:144]: The templates shouldn't be hard-coded, the code should list the folders inside `WebGLTemplates` and create the file in all folders that are actually there.

**Review by Arshia001 — CHANGES_REQUESTED** [review]

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs (hunk 2 above)
Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:144]: how do you mean?

Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:144]: or this 

```
string root = "Assets/WebGLTemplates/";
var info = new DirectoryInfo(root);
var fileInfo = info.GetFiles();
```

Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:144]: also if you noticed, both network files are different, so i'm not sure if a for x for each file work

**Comment by sneakzttv** [comment]
@juan Currently you'll see a newtonsoft error when you pull main, i have no idea why but this fixes it. @Arshia001 you may want to look into this too i'm not sure who or what caused it, this error was present when i started.

1: Create new references
image

2: remove unused references
image

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs (hunk 1 above)
Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:152]: fixed

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs [hunk 3]
```diff
@@ -43,6 +45,12 @@ void Awake()
             PlayerPrefs.Save();
         }
 
+        if (Token == ("Please Enter Your Chain's Native Token i.e Eth, Cro") && (PlayerPrefs.GetString("Token") != ""))
```
Comment (RyRy79261) [inline src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:48]: I think the terminology here is symbol, consider rephrasing to: 
```suggestion
        if (Symbol == ("Please Enter Your Chain's Native Symbol i.e Eth, Cro") && (PlayerPrefs.GetString("Symbol") != ""))
```

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs (hunk 2 above)
Comment (RyRy79261) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:144]: @sneakzttv @Arshia001 might mean with regards to `Web3GL-2020x`, there might be additional or different version templates in there, meaning this should be potentially created in all templates that appear to require the file, not just the one expected.

If I'm reading that correctly

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs (hunk 1 above)
Comment (RyRy79261) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs:152]: Not sure if I missed the fix, but I can suggest them here, would need to be done to label, token & rpc url probably 
```suggestion
        writer1.WriteLine("    id: " + PlayerPrefs.GetString("ChainID") + ",");
```

**Review by RyRy79261 — CHANGES_REQUESTED** [review]
A clean implementation besides a semantic question and indentation. But in terms of maintainability, why not create a `Class` for the server setting parameters, then serialize that directly into a file?

That way we're sticking to writing C# as much as possible and not json inside of string quotes

[Example 1](https://stackoverflow.com/questions/41427643/c-sharp-serialize-and-deserialize-json-to-txt-file)

[Example 2](https://stackoverflow.com/questions/41427643/c-sharp-serialize-and-deserialize-json-to-txt-file)

File: src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs (hunk 3 above)
Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs]: like that interms of indentation and symbol?

Comment (RyRy79261) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs]: @sneakzttv I mean the term for the chain's local token is usually referred to as "Symbol"

For example, the fields in Metamask
image

Comment (sneakzttv) [reply src/UnityPackage/Assets/Editor/ServerSettings/ChainsafeServerSettings.cs]: yah got it, updated the rest, left the object as is

**Review by juans-chainsafe — APPROVED** [review]
Tried a few prefabs with WebGL and Webwallet, the player settings where save dinto the network.js file correctly with the symbol!

**Review by KBryan — APPROVED** [review]

**Pull request merged**